crates/provider-blobstore-nats/src/blobstore.rs

```diff
@@ -0,0 +1,678 @@
+                chunk_size: Some(src_object.info.chunks),
+            };
+
+            // Put the object into the destination container
+            dst_container
+                .put(metadata, &mut src_object)
+                .await
+                .context("failed to copy object to destination container")
+                .map(|_| ())
+        }
+        .await
+        .map_err(|err| format!("{err:#}")))
+    }
+
+    // Move an object from one to the same (different key name, or revision number update), or another NATS Blobstore Container
+    #[instrument(level = "debug", skip(self))]
+    async fn move_object(
+        &self,
+        context: Option<Context>,
+        source: bindings::wrpc::blobstore::types::ObjectId,
+        destination: bindings::wrpc::blobstore::types::ObjectId,
+    ) -> anyhow::Result<Result<(), String>> {
+        Ok(async {
+            propagate_trace_for_ctx!(context);
+
+            // Skip the move if source and destination are the same
+            if source.container == destination.container && source.object == destination.object {
+                info!(
+                    "skipping moving object '{}' to itself in container '{}'",
+                    source.object, source.container
+                );
+                return Ok(());
+            }
+
+            // Retrieve the NATS blobstore connection
+            let blobstore = self
+                .get_blobstore(context)
+                .await
+                .context("failed to get NATS Blobstore connection")?;
+
+            // Get the source container (object store)
+            let src_container = blobstore
+                .jetstream
+                .get_object_store(&source.container)
+                .await
+                .context("failed to open source NATS Blobstore Container")?;
+
+            // Get the destination container (object store)
+            let dst_container = blobstore
+                .jetstream
+                .get_object_store(&destination.container)
+                .await
+                .context("failed to open destination NATS Blobstore Container")?;
+
+            // Get the source object
+            let mut src_object = src_container
+                .get(source.object.clone())
+                .await
+                .context("failed to read object from source container")?;
+
+            // Prepare metadata for the destination object
+            let metadata = async_nats::jetstream::object_store::ObjectMetadata {
+                name: destination.object.clone(),
+                description: src_object.info.description.clone(),
+                chunk_size: Some(src_object.info.chunks),
+            };
+
+            // Put the object into the destination container
+            dst_container
+                .put(metadata, &mut src_object)
+                .await
+                .context("failed to move object to destination container")?;
```

Comment:
Just want to confirm - if destination object exists, will this overwrite the destination object? 

Quote from the spec:

> overwrites destination object if it already existed.

Reply:
Yes, it does; that's NATS default behaviour.